Content.Shared/Roles/SharedRoleSystem.cs

```diff
@@ -202,47 +201,53 @@ private bool MindRolesUpdate(Entity<MindComponent?> ent)
             return false;
 
         //get the most important/latest mind role
-        var roleType = GetRoleTypeByTime(ent.Comp);
+        var role = GetCurrentRole(ent.Comp);
 
-        if (ent.Comp.RoleType == roleType)
+        if (ent.Comp.RoleType == role.Item1 &&  ent.Comp.Subtype == role.Item2)
             return false;
 
-        SetRoleType(ent.Owner, roleType);
+        SetRoleType(ent.Owner, role.Item1, role.Item2);
         return true;
     }
 
-    private ProtoId<RoleTypePrototype> GetRoleTypeByTime(MindComponent mind)
+    public (ProtoId<RoleTypePrototype>, LocId?) GetCurrentRole(MindComponent mind)
     {
-        // If any Mind Roles specify a Role Type, return the most recent. Otherwise return Neutral
+        return GetRoleTypeByTime(mind);
+    }
+
+    private (ProtoId<RoleTypePrototype>, LocId?) GetRoleTypeByTime(MindComponent mind)
+    {
+        // If any Mind Roles specify a Role Type, return the most recent one along with the subtype. Otherwise return Neutral
 
-        var roles = new List<ProtoId<RoleTypePrototype>>();
+        var roles = new List<(ProtoId<RoleTypePrototype>, LocId?)>();
 
         foreach (var role in mind.MindRoles)
         {
             var comp = Comp<MindRoleComponent>(role);
             if (comp.RoleType is not null)
-                roles.Add(comp.RoleType.Value);
+                roles.Add((comp.RoleType.Value, comp.Subtype));
         }
 
-        ProtoId<RoleTypePrototype> result = (roles.Count > 0) ? roles.LastOrDefault() : "Neutral";
+        var result = (roles.Count > 0) ? roles.LastOrDefault() : ("Neutral", null);
         return (result);
```

Comment:
You can just directly return the result. No need to put it in a variable which is only assigned once and then never changed.